### PR TITLE
Add formatting rules, and change rustfmt toolchain

### LINKED ISSUE #38: Add a rustfmt.toml to this project
Start by adding https://rust-lang.github.io/rustfmt/?version=v1.4.36&search=#imports_granularity

### PR DESCRIPTION
**Pull request by nyxkrage**
This closes #38 
formatting rules has been taken from my work.

### CODE REVIEW

**Review by satu — COMMENTED** [review]
I'd like to limit it for now to very few options, starting from the imports_granularity. I'll try locally and get back to you.

File: rust-toolchain [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[toolchain]
+channel="nightly"
+components=["rustfmt"]
+profile="default"
```
Comment (satu) [inline rust-toolchain]: newline at end of file pls

File: rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+error_on_unformatted=true
+fn_args_layout="Compressed"
+format_code_in_doc_comments=true
+format_macro_matchers=true
+format_strings=true
+hard_tabs=true
+match_arm_blocks=false
+match_block_trailing_comma=true
+max_width=120
+imports_granularity="Module"
```
Comment (satu) [inline rustfmt.toml:1]: please keep only this line and remove the rest. we'll iterate over this as we see fit

**Review by satu — CHANGES_REQUESTED** [review]
please revert the changes to the code and re-format. I'd like to keep the git blame history relevant hence we might have to be picky about adding new rustfmt standards.

**Comment by nyxkrage** [comment]
@satu Alright it should be fixed now

**Review by satu — APPROVED** [review]
🥇

**Pull request merged**